test/fail_compilation/check_param_inverted.d

```diff
@@ -0,0 +1,24 @@
+/* TEST_OUTPUT:
+---
+fail_compilation/check_param_inverted.d(21): Error: function `check_param_inverted.fun(int[] a, int b)` is not callable using argument types `(int, void[])`
+fail_compilation/check_param_inverted.d(21):        cannot pass argument `0` of type `int` to parameter `int[] a`
+fail_compilation/check_param_inverted.d(21):        would the parameters order be inverted ?
+fail_compilation/check_param_inverted.d(22): Error: function `check_param_inverted.lol(int[] a, int b, int c)` is not callable using argument types `(int, void[])`
+fail_compilation/check_param_inverted.d(22):        cannot pass argument `0` of type `int` to parameter `int[] a`
+fail_compilation/check_param_inverted.d(23): Error: template `check_param_inverted.cat` cannot deduce function from argument types `!()(int, char[][])`, candidates are:
+fail_compilation/check_param_inverted.d(17):        `check_param_inverted.cat(T : char[][], U : int)(T a, U b)`
+fail_compilation/check_param_inverted.d(23):        would the parameters order be inverted ?
+---
+*/
+module check_param_inverted;
+
+void fun(int[] a, int b){}
+void lol(int[] a, int b, int c){}
+void cat(T : char[][], U : int)(T a, U b){}
+
+void main()
+{
+    fun(0, []);
+    lol(0, []);
+    cat(0, new char[][](0));
```

Comment:
it would be good to add some UFCS checks too.

Reply:
and member functions.

Reply:
indeed, done.